### PR TITLE
Add 2026 year support with norms calculations

### LINKED ISSUE #350: Enable 2026 in fdm-app
## Overview

The application needs to support the year 2026. Currently, functionality is often limited to 2025. This issue covers enabling 2026 in the UI and setting up the initial calculation logic for 2026 norms (`gebruiksnormen`).

### 1. `fdm-calculator`: Implement 2026 Norms

The usage norms for 2026 should be based on the 2025 implementation, with the explicit removal of **Derogation**.

* **Duplicate & Rename:**
  * Copy the entire directory `src/norms/nl/2025` to `src/norms/nl/2026`.
  * Rename all files and functions within this new directory from `*2025*` to `*2026*` (e.g., `calculateNL2025StikstofGebruiksNorm` -> `calculateNL2026StikstofGebruiksNorm`).
  * Update internal imports and logic to reference 2026 data and dates.

* **Remove Derogation:**
  * In `src/norms/nl/2026/value/dierlijke-mest-gebruiksnorm.ts`, remove the logic that allows for derogation norms. The standard norm (170 kg N/ha) should be the only option for 2026.
  * Ensure any derogation-related inputs in `src/norms/nl/2026/value/input.ts` are either removed or ignored/defaulted to false.

* **Register New Norms:**
  * Update `src/norms/index.ts` to include a case for `year === "2026"` and wire up the new 2026 functions.

### 2. `fdm-app`: Enable 2026 in UI

Update the application to allow users to navigate to and configure data for 2026.

* **Norms Pages:**
  * Update `app/routes/farm.$b_id_farm.$calendar.norms._index.tsx` and `app/routes/farm.$b_id_farm.$calendar.norms.$b_id.tsx` to allow the `calendar` parameter to be "2026". Currently, it restricts access to 2025.
  * Update the "Go to 2025" button/logic to support navigating to 2026.

* **Calculator Integration:**
  * Update `app/integrations/calculator.ts`. The `createFunctionsForNorms` call is currently hardcoded to `("NL", "2025")`. Ensure it uses the correct year context or supports 2026.

* **Farm Creation & Settings:**
  * Update `app/routes/farm.create._index.tsx` to allow a start year of 2026.
  * **Derogation Settings:** Verify `app/routes/farm.$b_id_farm.settings.derogation.tsx`. It currently filters years `<= 2025`. **Do not change this.** This correctly prevents users from configuring derogation for 2026, aligning with the requirement to remove it.

* **Atlas:**
  * Update `app/components/blocks/atlas/atlas-url.ts` to extend the max year for map layers to 2026 (if applicable, or ensure safe fallback).

### 3. `fdm-core`

* **Derogation Validation:**
  * Retain the validation in `src/derogation.ts` that limits derogation years to `max 2025`. This serves as a backend guard against creating derogation records for 2026.

## Notes

* Other changes to 2026 norms that are not yet known will be handled in separate issues.

### PR DESCRIPTION
**Pull request by SvenVw**
## Summary by CodeRabbit

* **New Features**
  * Full 2026 support: nitrogen & phosphate norms, fertilizer‑filling calculators, data tables and input collection; farm creation now offers next‑year selection and year‑aware flows.
  * Memoized/cached calculation endpoints for faster responses.

* **Bug Fixes**
  * Korting edge case fixed for vanggewas sown Oct 15.
  * NL 335 (nature/heide) treated as non‑bouwland where applicable.

* **Tests**
  * Expanded test coverage for 2025 and 2026 norms and filling logic.

* **Documentation**
  * UI messages updated to show norms available for 2025 and 2026.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

Closes #350

### CODE REVIEW

File: fdm-calculator/src/norms/nl/2026/filling/input.test.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+            new Date(2026, 4, 15),
+        )
+        expect(getCultivations).toHaveBeenCalledWith(
+            mockFdm,
+            mockPrincipalId,
+            mockFieldId,
+            {
+                start: new Date(2026, 0, 1),
+                end: new Date(2026, 11, 31, 23, 59, 59, 999),
+            },
```
Comment (gerardhros) [inline fdm-calculator/src/norms/nl/2026/filling/input.test.ts]: is this date by purpose set up to miliseconds?

File: fdm-calculator/src/norms/nl/2026/filling/types.d.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    Fertilizer,
+    FertilizerApplication,
+    Field,
+} from "@svenvw/fdm-core"
+import type { RegionKey } from "../value/types"
+
+export type Table11Mestcodes = {
+    p_type_rvo: schema.fertilizersCatalogueTypeSelect["p_type_rvo"]
+    p_type_nitratesdirective: boolean
+    p_n_rt?: number
```
Comment (gerardhros) [inline fdm-calculator/src/norms/nl/2026/filling/types.d.ts]: what is meant with "p_type_nitratesdirective"?

**Review by gerardhros — APPROVED** [review]
LGTM. does this imply that all regulations for 2026 have been set or are you using 2025 as default and will replace optional changes in for example nv_areas later? 
a few minor questions to check. Well done.

File: fdm-calculator/src/norms/nl/2026/filling/input.test.ts (hunk 1 above)
Comment (SvenVw) [reply fdm-calculator/src/norms/nl/2026/filling/input.test.ts]: Yes to test the limit

File: fdm-calculator/src/norms/nl/2026/value/stikstofgebruiksnorm.ts [hunk 3]
```diff
@@ -0,0 +1,569 @@
...
+ *     The `getNormsForCultivation` helper function is called with the `selectedStandard`
+ *     and other relevant parameters (variety, cultivation end date).
+ *     This function applies a hierarchy of rules (sub-type periods, variety-specific,
+ *     derogation-specific) to return the precise `NormsByRegion` object for the cultivation.
+ *
+ * 7.  **Calculate Final Norm Value**:
+ *     From the `applicableNorms` object, the function retrieves the specific norms for the
+ *     determined `region`. The final `normValue` is then selected: if the field is in an
+ *     `is_nv_area`, the `nv_area` norm is used; otherwise, the `standard` norm is applied.
+ *
```
[the remark this answers is not in the document]
Comment (SvenVw) [reply fdm-calculator/src/norms/nl/2026/value/stikstofgebruiksnorm.ts]: No, currently the are no official changes known (except removal of derogation) to the Gebruiksnormen in 2026. When changes are known for 2026 we will update in a next PR.
